internal/defact/default.go

```diff
@@ -0,0 +1,262 @@
+package defact
+
+import (
+	"io/ioutil"
+	"os"
+	"path/filepath"
+	"regexp"
+	rt "runtime"
+	"strings"
+
+	"github.com/ActiveState/cli/internal/config"
+	"github.com/ActiveState/cli/internal/errs"
+	"github.com/ActiveState/cli/internal/fileutils"
+	"github.com/ActiveState/cli/internal/locale"
+	"github.com/ActiveState/cli/internal/logging"
+	"github.com/ActiveState/cli/internal/output"
+	"github.com/ActiveState/cli/internal/primer"
+	"github.com/ActiveState/cli/internal/strutils"
+	"github.com/ActiveState/cli/pkg/platform/runtime"
+	"github.com/gobuffalo/packr"
+	"github.com/thoas/go-funk"
+)
+
+type DefaultConfigurer interface {
+	Set(key string, value interface{})
+	GetString(key string) string
+}
+
+type exeFile struct {
+	fpath string
+	name  string
+	ext   string
+}
+
+func getDefaultProjectPath(d DefaultConfigurer) string {
+	return d.GetString("default_project_path")
+}
+
+func setDefaultProjectPath(d DefaultConfigurer, path string) {
+	d.Set("default_project_path", path)
+}
+
+type primable interface {
+	primer.Outputer
+}
+
+// isShimAndTargetsDir check if the specified filename is (probably) a shim and targets a file in the specified directory
+// This function is used during rollback to clean old shims
```

Comment:
Let's rephrase this to something like

```
checks if the given shim points to the given runtime directory
```

The only undetermined part is whether it points to our runtime directory. We own the shims and the shim directory, so there are no "maybe's" there.

Reply:
I think that it is always possible that the user adds their own binaries or other files there.  I guess it is not too bad to delete these files, as we "own" the directory, but I usually attempt to be "nice" in these cases. :)

Reply:
I don't think being "nice" helps us or the user here. If the user goes off the reservation then stuff will break. We shouldn't try to facilitate unsupported use-cases.